code/game/machinery/computer/arcade/orion.dm

```diff
@@ -0,0 +1,524 @@
+				to_chat(gamer, "<span class='userdanger'>Your body instantly contracts to that of one who has not eaten in months. Agonizing cramps seize you as you fall to the floor.</span>")
+		if(fuel <= 0)
+			reason = "You ran out of fuel, and drift, slowly, into a star."
+			if(obj_flags & EMAGGED)
+				gamer.adjust_fire_stacks(5)
+				gamer.IgniteMob() //flew into a star, so you're on fire
+				to_chat(gamer, "<span class='userdanger'>You feel an immense wave of heat emanate from the arcade machine. Your skin bursts into flames.</span>")
+
+	if(obj_flags & EMAGGED)
+		to_chat(gamer, "<span class='userdanger'>You're never going to make it to Orion...</span>")
+		gamer.death()
+		obj_flags &= ~EMAGGED //removes the emagged status after you lose
+		gameStatus = ORION_STATUS_START
+		name = "The Orion Trail"
+		desc = "Learn how our ancestors got to Orion, and have fun in the process!"
+
+	gamer?.mind?.adjust_experience(/datum/skill/gaming, 10)//learning from your mistakes is the first rule of roguelikes
+	return reason
+
+//Add Random/Specific crewmember
+/obj/machinery/computer/arcade/orion_trail/proc/add_crewmember(specific = "", update = TRUE)
+	var/newcrew = ""
+	if(specific)
+		newcrew = specific
+	else
+		if(prob(50))
+			newcrew = pick(GLOB.first_names_male)
+		else
+			newcrew = pick(GLOB.first_names_female)
+	if(newcrew)
+		settlers += newcrew
+		alive++
+	if(update)
+		new_settler_mood()//new faces!
+		update_static_data(usr)
+	return newcrew
+
+
+//Remove Random/Specific crewmember
+/obj/machinery/computer/arcade/orion_trail/proc/remove_crewmember(specific = "", dont_remove = "", update = TRUE)
+	var/list/safe2remove = settlers
+	var/removed = ""
+	if(dont_remove)
+		safe2remove -= dont_remove
+	if(specific && specific != dont_remove)
+		safe2remove = list(specific)
+	removed = pick(safe2remove)
+
+	if(removed)
+		if(lings_aboard && prob(40*lings_aboard)) //if there are 2 lings you're twice as likely to get one, obviously
+			lings_aboard = max(0,--lings_aboard)
+		settlers -= removed
+		alive--
+	if(update)
+		new_settler_mood()//bro, i...
+		update_static_data(usr)
+	return removed
+
+/**
+ * Crewmember executed code, targeted when there are no lings and untargeted when there are some
+ * If there was no suspected lings (aka random shots) this is just murder and counts towards killed crew
+ *
+ * Arguments:
+ * * gamer: carbon that may need emag effects applied
+ */
+/obj/machinery/computer/arcade/orion_trail/proc/execute_crewmember(mob/living/carbon/gamer, target)
+	var/sheriff = remove_crewmember(target) //I shot the sheriff
+	if(target)
+		killed_crew += 1 //if there was no suspected lings, this is just plain murder
+	playsound(loc,'sound/weapons/gun/pistol/shot.ogg', 100, TRUE)
+	if(!settlers.len || !alive)
+		say("The last crewmember [sheriff], shot themselves, GAME OVER!")
+		if(obj_flags & EMAGGED)
+			gamer.death()
+		set_game_over(gamer, "Your last pioneer committed suicide.")
+		if(killed_crew >= 4)
```

Comment:
This should use a define, or .len, or something to signify that its the number of crew you get.